examples/pytorch/mae.py

```diff
@@ -17,31 +28,33 @@ def __init__(self, vit):
 
         decoder_dim = 512
         self.mask_ratio = 0.75
-        self.patch_size = vit.patch_size
-        self.sequence_length = vit.seq_length
-        self.mask_token = nn.Parameter(torch.zeros(1, 1, decoder_dim))
-        self.backbone = masked_autoencoder.MAEBackbone.from_vit(vit)
-        self.decoder = masked_autoencoder.MAEDecoder(
-            seq_length=vit.seq_length,
-            num_layers=1,
-            num_heads=16,
-            embed_input_dim=vit.hidden_dim,
-            hidden_dim=decoder_dim,
-            mlp_dim=decoder_dim * 4,
-            out_dim=vit.patch_size**2 * 3,
-            dropout=0,
-            attention_dropout=0,
+        self.patch_size = vit.patch_embed.patch_size[0]
+
+        self.backbone = masked_vision_transformer_timm.MaskedVisionTransformerTIMM(
+            vit=vit
+        )
+        self.sequence_length = self.backbone.sequence_length
+        self.decoder = masked_autoencoder_timm.MAEDecoder(
```

Comment:
```suggestion
        self.backbone = MaskedVisionTransformerTIMM(
            vit=vit
        )
        self.sequence_length = self.backbone.sequence_length
        self.decoder = MAEDecoder(
```
Let's import the classes directly.

Please also change in the other files.